mono/metadata/debug-mono-ppdb.c

```diff
@@ -595,3 +596,130 @@ mono_ppdb_lookup_locals (MonoDebugMethodInfo *minfo)
 
 	return res;
 }
+
+/*
+* We use this to pass context information to the row locator
+*/
+typedef struct {
+	int idx;			/* The index that we are trying to locate */
+	int col_idx;		/* The index in the row where idx may be stored */
+	MonoTableInfo *t;	/* pointer to the table */
+	guint32 result;
+} locator_t;
+
+static int
+table_locator (const void *a, const void *b)
+{
+	locator_t *loc = (locator_t *)a;
+	const char *bb = (const char *)b;
+	guint32 table_index = (bb - loc->t->base) / loc->t->row_size;
+	guint32 col;
+
+	col = mono_metadata_decode_row_col(loc->t, table_index, loc->col_idx);
+
+	if (loc->idx == col) {
+		loc->result = table_index;
+		return 0;
+	}
+	if (loc->idx < col)
+		return -1;
+	else
+		return 1;
+}
+
+typedef gboolean (*GuidComparer) (guint8* guid);
+static gboolean
+is_async_method_stepping_information_guid (guint8* debug_info_kind_guid)
+{
+	return debug_info_kind_guid [0] == 0xC5 &&
```

Comment:
Would be better to declare this as a static guint8 array and compare using a loop.